nvme/src/types.rs

```diff
@@ -137,6 +137,16 @@ pub enum NVMeError {
         max_append: u32,
         trying_to_append: u32,
     },
+    ExtraContext{
+        context: String,
+        original_error: Box<NVMeError>
+    }
+}
+
+impl NVMeError {
+    pub fn add_context(&mut self, context: String) {
+        *self = NVMeError::ExtraContext { context, original_error: Box::new(self.clone()) }
```

Comment:
What is the reason behind doing this for adding extra information? Couldn't we just have an appropriate error?